### PR TITLE
P2P package cleanup

### PR DESCRIPTION
**Pull request by iulianpascalau**
- refactor p2p package

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

File: p2p/libp2p/metrics/disabledConnectionsWatcher_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDisabledConnectionsWatcher_MethodsShouldNoptPanic(t *testing.T) {
```
Comment (bogdan-rosianu) [inline p2p/libp2p/metrics/disabledConnectionsWatcher_test.go:11]: Nopt -> Not

File: p2p/libp2p/metrics/errors.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package metrics
+
+import "errors"
+
+var errInvalidTimeToLive = errors.New("invalid time to live")
```
Comment (bogdan-rosianu) [inline p2p/libp2p/metrics/errors.go:5]: might rename this error

File: p2p/libp2p/metrics/factory/connectionWatcherFactory.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const disabledConnectionsWatcher = "disabled"
+
+// NewConnectionsWatcher creates a new ConnectionWatcher instance based on the input parameters
+func NewConnectionsWatcher(connectionsWatcherType string, timeToLive time.Duration) (p2p.ConnectionsWatcher, error) {
+	switch connectionsWatcherType {
+	case printConnectionsWatcher:
+		return metrics.NewPrintConnectionsWatcher(timeToLive)
+	case disabledConnectionsWatcher:
+		return metrics.NewDisabledConnectionsWatcher(), nil
+	default:
```
Comment (bogdan-rosianu) [inline p2p/libp2p/metrics/factory/connectionWatcherFactory.go:21]: I would return the disabled connections watcher if the config's value isn't right

File: p2p/libp2p/metrics/factory/errors.go [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package factory
+
+import "errors"
+
+var errUnkownConnectionWatcherType = errors.New("unknown connection type")
```
Comment (bogdan-rosianu) [inline p2p/libp2p/metrics/factory/errors.go:5]: Unkown -> Unknown

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: integrationTests/p2p/peerDisconnecting/peerDisconnecting_test.go [hunk 5]
```diff
@@ -67,12 +70,12 @@ func testPeerDisconnectionWithOneAdvertiser(t *testing.T, p2pConfig config.P2PCo
...
 		SyncTimer:            &testscommon.SyncTimerStub{},
 	}
-	//Step 1. Create advertiser
+	// Step 1. Create advertiser
 	advertiser, err := libp2p.NewMockMessenger(argSeeder, netw)
 	require.Nil(t, err)
 	p2pConfig.KadDhtPeerDiscovery.InitialPeerList = []string{integrationTests.GetConnectableAddress(advertiser)}
 
-	//Step 2. Create noOfPeers instances of messenger type and call bootstrap
+	// Step 2. Create noOfPeers instances of messenger type and call bootstrap
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/p2p/peerDisconnecting/peerDisconnecting_test.go:78]: done

File: p2p/libp2p/connectionMonitor/libp2pConnectionMonitorSimple.go [hunk 6]
```diff
@@ -77,9 +84,10 @@ func (lcms *libp2pConnectionMonitorSimple) doReconn() {
 }
 
 // Connected is called when a connection opened
-func (lcms *libp2pConnectionMonitorSimple) Connected(netw network.Network, _ network.Conn) {
+func (lcms *libp2pConnectionMonitorSimple) Connected(netw network.Network, conn network.Conn) {
 	allPeers := netw.Peers()
 
+	lcms.connectionsWatcher.NewKnownConnection(core.PeerID(conn.RemotePeer()), conn.RemoteMultiaddr().String())
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply p2p/libp2p/connectionMonitor/libp2pConnectionMonitorSimple.go:90]: I do not think that is required. the caller code looks like
```GO
//....
// Add two swarm refs:
	// * One will be decremented after the close notifications fire in Conn.doClose
	// * The other will be decremented when Conn.start exits.
	s.refs.Add(2)

	// Take the notification lock before releasing the conns lock to block
	// Disconnect notifications until after the Connect notifications done.
	c.notifyLk.Lock()
	s.conns.Unlock()

	s.notifyAll(func(f network.Notifiee) {
		f.Connected(s, c)
	})
	c.notifyLk.Unlock()

	c.start()
```
the code will fail on `c.notifyLk.Lock()` sooner than my code

File: p2p/libp2p/discovery/factory/peerDiscovererFactory.go [hunk 7]
```diff
@@ -17,46 +17,46 @@ const defaultSeedersReconnectionInterval = time.Minute * 5
 
 var log = logger.GetOrCreate("p2p/discovery/factory")
 
+// ArgsPeerDiscoverer is the DTO struct
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply p2p/libp2p/discovery/factory/peerDiscovererFactory.go:20]: done

File: p2p/libp2p/metrics/disabledConnectionsWatcher_test.go (hunk 1 above)
Comment (iulianpascalau) [reply p2p/libp2p/metrics/disabledConnectionsWatcher_test.go:11]: done

File: p2p/libp2p/metrics/errors.go (hunk 2 above)
Comment (iulianpascalau) [reply p2p/libp2p/metrics/errors.go:5]: renamed

File: p2p/libp2p/metrics/factory/connectionWatcherFactory.go (hunk 3 above)
Comment (iulianpascalau) [reply p2p/libp2p/metrics/factory/connectionWatcherFactory.go:21]: might hide improper configuration leading to missing or badly configured p2p.toml file

File: p2p/libp2p/metrics/factory/errors.go (hunk 4 above)
Comment (iulianpascalau) [reply p2p/libp2p/metrics/factory/errors.go:5]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by liviuancas-elrond — APPROVED** [review]

**Pull request merged**